### ISSUE TITLE
AWS DataSync SMB locations

### ISSUE THREAD

**Issue by ewbankkit** [terraform-providers/terraform-provider-aws#9868]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS DataSync now supports the Server Message Block (SMB) protocol.

### New or Affected Resource(s)

* `aws_datasync_location_smb`

### Potential Terraform Configuration

```hcl
resource "aws_datasync_location_smb" "example" {
  agent_arns = ["${aws_datasync_agent.example.arn}"]

  server_hostname = "smb.example.com"
  subdirectory    = "/exported/path"
}
```

### References

[Announcement](https://aws.amazon.com/about-aws/whats-new/2019/08/aws-datasync-can-now-transfer-data-to-and-from-smb-files-shares/).
[API reference](https://docs.aws.amazon.com/datasync/latest/userguide/API_CreateLocationSmb.html).

Requires:
* https://github.com/terraform-providers/terraform-provider-aws/pull/9770

**Comment by cjbeckwith** [comment]
Curious where you are at in regards to the aws_datasync_location_smb resource.  I see that AWS PS just released a cmdlet for it, but as of 2.31 I don't see reference to this particular resource.  Looking to use it sooner rather than later.

**Comment by mark-bixler** [comment]
How many votes are needed before this becomes priority? :) Tasked with a massive project to roll-out 27 agents and various tasks to SMB Share

**Comment by jonsherrard** [comment]
Hello there, very keen on using this SMB data sync resource.

I'm happy to have a run at doing the work, is there a de facto set of instructions for contributing to Terraform?

**Comment by thatderek** [comment]
@mark-bixler If you want to go upvote #10381, that one is ready for review and your :+1: might help it get visibility.

**Comment by bflad** [comment]
Support for this new resource has been merged and will release with version 2.48.0 of the Terraform AWS Provider, Thursday this week. Thanks to @thatderek for the implementation. 👍
